### PR TITLE
feat(composer): commit vendor changes for composer manager

### PR DESCRIPTION
**Pull request by esetnik**
## Changes:
Implements feature request #7661 

## Context:
This is an implementation which follows the `gomod` manager behavior of including changes to the `vendor` folder in the PR that renovate generates for composer dependencies.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/composer/artifacts.ts [hunk 1]
```diff
@@ -110,7 +110,9 @@ export async function updateArtifacts({
     logger.debug('No composer.lock found');
     return null;
   }
-  await ensureLocalDir(getSiblingFileName(packageFileName, 'vendor'));
+
+  const vendorDir = getSiblingFileName(packageFileName, 'vendor');
+  await ensureLocalDir(vendorDir);
```
Comment (rarkins) [inline lib/manager/composer/artifacts.ts:115]: I actually forget why we create the `vendor/` dir. It could possibly be for legacy reasons. Fine to leave like this for backwards-compatibility reasons.

**Review by rarkins — COMMENTED** [review]
This looks good, but can you test it against a real repository to confirm it works as expected?

File: lib/manager/composer/artifacts.spec.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import { exec as _exec } from 'child_process';
+import _fs from 'fs-extra';
 import { join } from 'upath';
 import { envMock, mockExecAll } from '../../../test/execUtil';
-import { fs, git, mocked } from '../../../test/util';
+import { git, mocked } from '../../../test/util';
```
Comment (viceice) [inline lib/manager/composer/artifacts.spec.ts:5]: Why this change? why do you moch `fs-extra` instead of our `fs` wrapper?

Comment (esetnik) [reply lib/manager/composer/artifacts.spec.ts:5]: @viceice I followed the example of `gomod` testing which appeared to be more recent. See https://github.com/renovatebot/renovate/blob/master/lib/manager/gomod/artifacts.spec.ts#L14

**Comment by esetnik** [comment]
> This looks good, but can you test it against a real repository to confirm it works as expected?

I ran it against a test repository and have produced the following PRs https://github.com/esetnik/composer-vendor-test/pull/2 https://github.com/esetnik/composer-vendor-test/pull/1

File: lib/manager/composer/artifacts.spec.ts (hunk 2 above)
Comment (viceice) [reply lib/manager/composer/artifacts.spec.ts:5]: I think that's wrong, can you please revert to mock `util/fs` again. We need to change gomod later too.

**Review by viceice — COMMENTED** [review]
Just revert the fs-extra mock please, otherwise LGTM

**Review by rarkins — CHANGES_REQUESTED** [review]
Revert fs mocking changes as per @viceice comments

**Review by rarkins — APPROVED** [review]
